picocrt/crt0.h

```diff
@@ -107,16 +114,21 @@ __start(void)
 	__libc_init_array();
 #endif
 
-#if defined(CRT0_SEMIHOST) && defined(__ARM_SEMIHOST)
+#if defined(CRT0_SEMIHOST) &&                                                  \
+    (defined(__ARM_SEMIHOST) || defined(__HEXAGON_ARCH__))
 #define CMDLINE_LEN     1024
 #define ARGV_LEN        64
         static char cmdline[CMDLINE_LEN];
         static char *argv[ARGV_LEN];
         int argc = 0;
 
-        if (sys_semihost_get_cmdline(cmdline, sizeof(cmdline)) == 0 &&
-            cmdline[0])
-        {
+        int rc;
+#ifdef __ARM_SEMIHOST
+        rc = sys_semihost_get_cmdline(cmdline, sizeof(cmdline));
+#elif defined(__HEXAGON_ARCH__)
+        rc = get_cmdline(cmdline, sizeof(cmdline));
```

Comment:
Please just rename the hexagon version to `sys_semihost_get_cmdline` and leave everything else alone; that way we won't add `get_cmdline` to the C library namespace, potentially conflicting with something from the application. At some point, we should rename all of these `sys_semihost_*` functions to something in the private C library namespace, perhaps `__semihost_*` instead. But for now, let's just use the same name in both hexagon and arm-compatible implementations.